File: src/notification/Notification.js
import React, {Component} from 'react';

class Notification extends Component {

    constructor() {
        super();

        this._showNotification = this._showNotification.bind(this);
    }


    _showNotification() {
        if(this.props.noOfComments > 2) {
            return <h1>Too many Comments. Reduce those</h1>
        }
    }

    render() {
        return (
            <div className="notification">
                {
                    this._showNotification()
                }
            </div>
        );
    };


}

export default Notification;

File: src/main/scala/samples/r4s/Notification.scala
package samples.r4s

import com.github.ahnfelt.react4s._

case class Notification() extends Component[NoEmit] {
  val commentStore = CommentStore(this)

  override def render(get: Get): Element = E.div(
    if (get(commentStore).length > 2) {
      E.h1(Text("Too many Comments. Reduce those"))
    } else {
      E.h1()
    }
  )
}
